### PR TITLE
Display devices and simulators in separate sections

### PR DESCRIPTION
**Pull request by lfroms**
### What does this change accomplish?

This change updates how devices are displayed in the Tophat menu.  Previously, devices and pinned simulators specifically were displayed under a single "Devices" section, with all remaining simulators in an "Other Devices" section.

Now, devices get their own section, pinned simulators get their own section, and the remaining simulators are displayed in an "Other Simulators" section.  This should make the pinning mechanism slightly less confusing.

Also, the "Other Simulators" section changes names to "Simulators" when no simulators are pinned to avoid confusion.

### How have you achieved it?

By reorganizing the UI code that produces these menus, taking the opportunity to convert `DeviceManager` to use the new Observation framework rather than `ObservableObject`.

### How can the change be tested?

Open the Tophat menu and ensure that you can interact with devices and simulators as per usual, but that they are organized into separate sections.

### CODE REVIEW

File: Tophat/Views/DeviceList.swift [hunk 1]
```diff
@@ -11,43 +11,50 @@ import TophatFoundation
...
+			) {
 				ScrollView(.vertical, showsIndicators: false) {
-					DevicePicker(devices: secondaryDevices)
+					DevicePicker(devices: otherSimulators)
 				}
 				.frame(maxHeight: 240)
 			}
 		}
-		.onChange(of: scenePhase) { newValue in
+		.onChange(of: scenePhase) { oldValue, newValue in
```
Comment (Caiopia) [inline Tophat/Views/DeviceList.swift:57]: Is this here just to fix one of the warnings we had?

File: Tophat/Models/AppleApplication.swift [hunk 2]
```diff
@@ -68,8 +67,24 @@ struct AppleApplication: Application {
...
+
+		for platformName in platformNames {
+			switch platformName {
+				case "iPhoneOS", "iPhoneSimulator":
+					return .iOS
+				case "WatchOS", "WatchSimulator":
+					return .watchOS
+				case "AppleTVOS", "AppleTVSimulator":
+					return .tvOS
+				default:
```
Comment (Caiopia) [inline Tophat/Models/AppleApplication.swift:82]: What about `VisionOS`? 😛

File: Tophat/Views/DeviceList.swift (hunk 1 above)
Comment (lfroms) [reply Tophat/Views/DeviceList.swift:57]: Yes indeed!

**Review by Caiopia — APPROVED** [review]

File: Tophat/Models/AppleApplication.swift [hunk 3]
```diff
@@ -68,8 +67,24 @@ struct AppleApplication: Application {
 	}
 
 	var platform: Platform {
-		// Only iOS supported, for now.
-		.iOS
+		guard let platformNames = bundle?.infoDictionary?["CFBundleSupportedPlatforms"] as? [String] else {
```
Comment (Caiopia) [inline Tophat/Models/AppleApplication.swift:70]: At first I thought this was Tophat's supported platforms, which would only be `macOS`, but I see now it's the downloaded app's supported platforms. 👌

Comment (lfroms) [reply Tophat/Models/AppleApplication.swift:70]: Yeah, we create a `Bundle` for the downloaded app to read it as well.

File: Tophat/Models/AppleApplication.swift (hunk 2 above)
Comment (lfroms) [reply Tophat/Models/AppleApplication.swift:82]: Going to pull that change into a separate branch, there's a few more files to change.

**Pull request merged**